pkg/test/echo/server/forwarder/protocol.go

```diff
@@ -92,14 +132,15 @@ func newProtocol(cfg Config) (protocol, error) {
 		if cfg.Request.Http2 && scheme.Instance(u.Scheme) == scheme.HTTPS {
 			proto.client.Transport = &http2.Transport{
 				TLSClientConfig: &tls.Config{
-					InsecureSkipVerify: true,
+					GetClientCertificate: getClientCertificate,
+					InsecureSkipVerify:   true,
 				},
 				DialTLS: func(network, addr string, cfg *tls.Config) (net.Conn, error) {
 					return tls.Dial(network, addr, cfg)
 				},
 			}
 		} else if cfg.Request.Http2 {
-			proto.client.Transport = &http2.Transport{
+			transport := &http2.Transport{
```

Comment:
this is supposed to be plaintext. L132 handles HTTPS

Reply:
Removed.